src/meds_torch/data/components/pytorch_dataset.py

```diff
@@ -295,19 +289,32 @@ def load_subject(
         shapes.
 
         Notes:     - This method applies sequence length constraints and sampling strategies       as
-        specified in the dataset configuration.     - It handles different collate types (triplet, etc.)
-        differently.     - The method is decorated with @SeedableMixin.WithSeed to ensure
-        reproducibility.
+        specified in the dataset configuration.     - The method is decorated with @SeedableMixin.WithSeed to
+        ensure reproducibility.
         """
         shard = self.subj_map[subject_id]
         subject_idx = self.subj_indices[subject_id]
         static_row = self.static_dfs[shard][subject_idx].to_dict()
 
+        max_seq_len = self.config.max_seq_len
+
         out = {
             "static_indices": static_row["static_indices"].item().to_list(),
             "static_values": static_row["static_values"].item().to_list(),
         }
 
+        if self.config.do_prepend_static_data:
+            n_static = len(out["static_indices"])
+            if n_static >= max_seq_len:
+                raise ValueError(
+                    f"Static data length {n_static} matches or exceeds "
+                    f"max_seq_len {max_seq_len} for subject {subject_id}!"
+                )
+
+            max_seq_len -= n_static
+        if self.config.postpend_eos_token:
+            max_seq_len -= 1
```

Comment:
This logic should by handled by the hydra resolver for the max_seq_len: https://github.com/Oufattole/meds-torch/blob/06e38f05901598ec4b85cc3221af739e445bb98e/src/meds_torch/utils/resolvers.py#L25

Reply:
Ok -- I think it makes more sense to handle it here, rather than there, because if the user sets max seq len to 256, that may be because their GPU memory is limited and they only want that sequence length being run through attention on the GPU; adjusting the max seq length in the hydra resolver breaks that API constraint. Do you agree @Oufattole?

Reply:
Oh yeah, you're totally right, thanks for explaining!